securedrop/loaddata.py

```diff
@@ -0,0 +1,419 @@
+    source = Source(filesystem_id, journalist_designation)
+    source.pending = False
+    db.session.add(source)
+    db.session.commit()
+
+    # Create source directory in store
+    os.mkdir(current_app.storage.path(source.filesystem_id))
+
+    # Generate source key
+    current_app.crypto_util.genkeypair(source.filesystem_id, codename)
+
+    return source, codename
+
+
+def star_source(source: Source) -> None:
+    """
+    Adds a SourceStar record for the source.
+    """
+    star = SourceStar(source, True)
+    db.session.add(star)
+    db.session.commit()
+
+
+def load(args: argparse.Namespace) -> None:
+    """
+    Populate the database.
+    """
+    if not os.environ.get("SECUREDROP_ENV"):
+        os.environ["SECUREDROP_ENV"] = "dev"
+
+    app = journalist_app.create_app(config)
+    with app.app_context():
+        try:
+            default_journalist = add_journalist("journalist", is_admin=True)
+        except IntegrityError as e:
+            db.session.rollback()
+            if "UNIQUE constraint failed: journalists." in str(e):
+                default_journalist = Journalist.query.filter_by(username="journalist").one()
+            else:
+                raise e
+
+        try:
+            dellsberg = add_journalist("dellsberg")
+        except IntegrityError as e:
+            db.session.rollback()
+            if "UNIQUE constraint failed: journalists." in str(e):
+                dellsberg = Journalist.query.filter_by(username="dellsberg").one()
+            else:
+                raise e
+
+        try:
+            journalist_to_be_deleted = add_journalist(
+                username="clarkkent", first_name="Clark", last_name="Kent"
+            )
+        except IntegrityError as e:
+            db.session.rollback()
+            if 'UNIQUE constraint failed: journalists.' in str(e):
+                journalist_to_be_deleted = Journalist.query.filter_by(username="clarkkent").one()
+            else:
+                raise e
+
+        for i in range(1, args.journalist_count + 1):
+            add_journalist()
+
+        starred_sources_count = int(args.source_count * args.source_star_fraction)
+        replied_sources_count = int(args.source_count * args.source_reply_fraction)
+
+        for i in range(1, args.source_count + 1):
+            source, codename = add_source()
+
+            for _ in range(args.messages_per_source):
+                journalist_who_saw = random.choice(
+                    [default_journalist, dellsberg, None]
+                )
+                submit_message(source, journalist_who_saw)
+
+            for _ in range(args.files_per_source):
+                journalist_who_saw = random.choice(
+                    [default_journalist, dellsberg, None]
+                )
+                submit_file(source, journalist_who_saw)
+
+            if i <= starred_sources_count:
+                star_source(source)
+
+            reply_to_source = i <= replied_sources_count
```

Comment:
@creviera - this is why no replies after a certain point (can get 100% by setting reply fraction arg to 1)

Reply:
good find